src/binfhe/lib/binfhecontext.cpp

```diff
@@ -42,92 +42,214 @@ void BinFHEContext::GenerateBinFHEContext(uint32_t n, uint32_t N,
                                           const NativeInteger &Q, double std,
                                           uint32_t baseKS, uint32_t baseG,
                                           uint32_t baseR, BINFHEMETHOD method) {
-  auto lweparams = std::make_shared<LWECryptoParams>(n, N, q, Q, std, baseKS);
+  auto lweparams = std::make_shared<LWECryptoParams>(n, N, q, Q, Q, std, baseKS);
   m_params =
       std::make_shared<RingGSWCryptoParams>(lweparams, baseG, baseR, method);
 }
 
+
+
+void BinFHEContext::GenerateBinFHEContext(BINFHEPARAMSET set,
+                                          bool arbFunc, uint32_t logQ, long N,  
+                                          BINFHEMETHOD method, bool timeOptimization) {
+  
+  if(method == AP){
+    std::string errMsg =
+             "Currently only support for method = GINX.";
+    PALISADE_THROW(not_implemented_error, errMsg);
+  }
+  if(set != STD128 && set != TOY){
+    std::string errMsg =
+             "Currently only support for STD128 and TOY.";
+    PALISADE_THROW(not_implemented_error, errMsg);
+  }
+
+  SecurityLevel  sl = HEStd_128_classic;
+
+  m_timeOptimization = timeOptimization;
+  auto logQprime = 27;
+  uint32_t baseG = 0;
+  if(logQ == 11){
+    baseG = 1<<5;
+  } else if (logQ > 29) {
+    std::string errMsg =
+             "Error: Don't support logQ > 29.";
+    PALISADE_THROW(not_implemented_error, errMsg);
+    return;
+  } else if (logQ > 25) {
+    baseG = 1<<14;
+    logQprime = 54;
+  } else if (logQ > 16) {
+    baseG = 1<<18;
+    logQprime = 54;
+  } else if (logQ > 11) {
+    baseG = 1<<27;
+    logQprime = 54;
+  } else {
+    std::string errMsg =
+              "Error: Don't support logQ < 11.";
+    PALISADE_THROW(not_implemented_error, errMsg);
+    return;
+  }
+
+  uint32_t ringDim = StdLatticeParm::FindRingDim(HEStd_ternary,sl,logQprime); // choose minimum ringD satisfying sl and Q
+  if (N >= ringDim){ // if specified some larger N, security is also satisfied
+    ringDim = N;
+  }
+  NativeInteger Q = PreviousPrime<NativeInteger>(FirstPrime<NativeInteger>(logQprime, ringDim), ringDim); // find prime Q for NTT
+  long q = ringDim * 2; // q = N*2 by default for maximum plaintext space
```

Comment:
"long q" should be assigned using a ternary operator ?: as the condition is simple and the resulting line will be short